### PR TITLE
CreateProcess misc fixes

### PR DESCRIPTION
**Pull request by bmiklautz**
* fix setting of default signal handler
* fix faulty check
* set the current working directory even if no token was set

### CODE REVIEW

**Comment by akallabeth** [comment]
+1, LGTM

File: winpr/libwinpr/thread/process.c [hunk 1]
```diff
@@ -317,11 +317,12 @@ BOOL _CreateProcessExA(HANDLE hToken, DWORD dwLogonFlags,
...
 				setuid((uid_t) token->UserId);
 
-			/* TODO: add better cwd handling and error checking */
-			if (lpCurrentDirectory && strlen(lpCurrentDirectory) > 0)
-				chdir(lpCurrentDirectory);
 		}
 
+		/* TODO: add better cwd handling and error checking */
+		if (lpCurrentDirectory && strlen(lpCurrentDirectory) > 0)
+			chdir(lpCurrentDirectory);
```
Comment (bmiklautz) [inline winpr/libwinpr/thread/process.c]: Since @akallabeth mentioned it the chdir at this point is unchecked on purpose.

**Comment by hardening** [comment]
+1

**Comment by nfedera** [comment]
+1 thanks!!

**Pull request merged**